R/grouped_epi_archive.R

```diff
@@ -280,16 +281,18 @@ grouped_epi_archive =
               if (! (is.atomic(comp_value) || is.data.frame(comp_value))) {
                 Abort("The slide computation must return an atomic vector or a data frame.")
               }
+              # Convert from data.frame to tibble for speed.
+              # Label every result row with the `ref_time_value`
+              res <- as_tibble(data.frame(time_value = ref_time_value))
```

Comment:
IIRC `as_tibble` can also be a good amount of overhead in a tight loop.  Maybe see if this can be done after collecting the different results?

Reply:
Also, maybe we should consider removing `%>%`s within loops.  Probably still can't use `|>` yet though.  Only time when our loops are tight enough that this might matter is in latency analysis and MAYBE lag extraction (though snapshot completion probably makes it not a tight loop in the latter case).

Reply:
> maybe we should consider removing %>%s within loops

I can add this in. You're imagining changes like `select(...) %>% filter(...)` -> `filter(select(...), ...)` for now? At least where it doesn't impact readability.

Reply:
> `as_tibble`... can be done after collecting the different results?

I can't find a nice way. I changed this section a bit to make it faster and, even though it still calls `as_tibble`, is about as fast in my benchmarking and profiling as a janky non-`as_tibble` approach. Will leave this as-is for now.

Reply:
(Another approach creating a new `data.frame` every iteration was about as fast in benchmarking as the current version.)

Reply:
> > `as_tibble`... can be done after collecting the different results?
> 
> I can't find a nice way. I changed this section a bit to make it faster and, even though it still calls `as_tibble`, is about as fast in my benchmarking and profiling as a janky non-`as_tibble` approach (could be different with a different test query, of course). Will leave this as-is for now.

Yes, something like that.  Or maybe, for easier readability, though I don't particularly like this pattern due to making debugging harder (and it might actually garbage-collect differently, due to a potentially higher refcount and potentially setting the "named" garbage collection flag whatever that does):
```r
tmp <- `select(...)`
tmp <- `filter(tmp, ...)`
```

Reply:
Another approach?: I think there may actually be something like `new_tibble` or `new_data_frame` or other related constructs for disabling some validation.

Reply:
So the alternate approach I came up with was

```
df <- data.frame(time_value = ref_time_value)
df[[new_col]] <- I(list(comp_value_raw))
return(df)
```

and then do the rbindlist -> as_tibble outside the loop. So there is no tibble conversion inside the loop and we still get our loop output col in list format. At least in benchmarking, this wasn't quite as fast as what I have in the code now (list -> as_tibble in loop), but that wouldn't necessarily be true in real calls. `new_tibble` is faster, but do we then need to run `validate_tibble` as well?

Reply:
(At this point, this step isn't taking that much time, esp compared to the group_by-group_modify in the loop, so we don't need to think about it super in-depth)

Reply:
Added `validate_tibble` in for safety. It makes using `new_tibble` slower, but still faster than other methods.